src/rules/pydocstyle/rules/multi_line_summary_start.rs

```diff
@@ -27,21 +30,58 @@ pub fn multi_line_summary_start(checker: &mut Checker, docstring: &Docstring) {
             .rules
             .enabled(&Rule::MultiLineSummaryFirstLine)
         {
-            checker.diagnostics.push(Diagnostic::new(
+            let mut diagnostic = Diagnostic::new(
                 violations::MultiLineSummaryFirstLine,
                 Range::from_located(docstring.expr),
-            ));
+            );
+            if checker.patch(diagnostic.kind.rule()) {
+                let loc = docstring.expr.location;
+                let mut endl = loc.row() + 1;
+                // Delete until first non-whitespace char.
+                for line in content_lines {
+                    if let Some(endc) = line.find(|c: char| !c.is_whitespace()) {
+                        let start = Location::new(loc.row(), loc.column() + first_line.len());
+                        let end = Location::new(endl, endc);
+                        diagnostic.amend(Fix::deletion(start, end));
```

Comment:
This could cause exceeding `line-length` on the start row. Should that be taken into account somehow, e.g. skipping the fix if it'd happen?

Reply:
We do sometimes check that the generated line is `< checker.settings.line_length`. But mostly for opinionated checks, like proposed simplifications. In this case, I think it's okay to apply the fix anyway. It's consistent with how we handle other docstring fixes.